nntrainer/tensor/tensor_wrap_specs.h

```diff
@@ -35,19 +36,22 @@ enum class WeightRegularizer {
  *
  */
 enum class TensorLifespan {
-  UNMANAGED = 0b0, /**< tensor with no lifespan, will not be allocated */
-  FORWARD_FUNC_LIFESPAN = 0b01,  /**< tensor must not be reset before during the
-                            forward function call, eg. temporary tensors
-                            needed during forward operations */
-  BACKWARD_FUNC_LIFESPAN = 0b10, /**< tensor must not be reset before during the
-                            backward function call, eg. temporary tensors
+  UNMANAGED = 0b000, /**< tensor with no lifespan, will not be allocated */
+  FORWARD_FUNC_LIFESPAN = 0b001, /**< tensor must not be reset before during the
+                           forward function call, eg. temporary tensors
+                           needed during forward operations */
+  CALC_DERIV_LIFESPAN = 0b010,
+  CALC_GRAD_LIFESPAN = 0b100,
+  BACKWARD_FUNC_LIFESPAN = 0b110, /**< tensor must not be reset before during
```

Comment:
CALC_GRAD_DERIV_LIFESPAN?

Reply:
BACKWARD_FUNC_LIFESPAN is already in use so I guess this has to be done with in a separate PR, if we mean to.

Reply:
I updated, and made an alias for the backward compat